### PR TITLE
Java CUDA 12 support

### PR DESCRIPTION
**Pull request by jchen351**
### Description

- This PR combine all CUDA 12 stage into the Zip-nuget-... pipeline.
- It also enables the cuda12 support

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 1]
```diff
@@ -99,49 +99,23 @@ variables:
   ${{ if eq(parameters.CudaVersion, '12.2') }}:
     value: 12.4
 
-stages:
-- stage: Setup
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:103]: This part has been moved to setup-variables-stage.yml

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 2]
```diff
@@ -164,38 +138,6 @@ stages:
     - template: templates/component-governance-component-detection-steps.yml
       parameters :
         condition : 'succeeded'
-
-
-- stage: Download_Java_Tools
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:169]: This part hase been moved to 
tools/ci_build/github/azure-pipelines/stages/jobs/steps/download-java-tools-step.yml

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 3]
```diff
@@ -231,41 +173,6 @@ stages:
     buildJava: true
     buildNodejs: true
 
-#CUDA without tensorrt
-- template: templates/win-ci.yml
-  parameters:
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:236]: This part has been moved to `tools/ci_build/github/azure-pipelines/stages/nuget-win-cuda-packaging-stage.yml`

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 4]
```diff
@@ -332,503 +239,31 @@ stages:
         condition: 'succeeded'
     - template: templates/clean-agent-build-directory-step.yml
 
-- stage: Jar_Packaging_GPU
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:335]: This stage has been moved to `stages/java-cuda-packaging-stage.yml`

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 5]
```diff
@@ -332,503 +239,31 @@ stages:
...
-
-    - template: templates/component-governance-component-detection-steps.yml
-      parameters:
-        condition: 'succeeded'
-    - task: mspremier.PostBuildCleanup.PostBuildCleanup-task.PostBuildCleanup@3
-      displayName: 'Clean Agent Directories'
-      condition: always()
-
-
-- stage: Windows_Packaging_combined_GPU
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:519]: This stage has been moved to stages/nuget-win-cuda-packaging-stage.yml

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 6]
```diff
@@ -332,503 +239,31 @@ stages:
...
-      inputs:
-        artifactName: 'onnxruntime-win-x64-gpu'
-        targetPath: '$(Build.ArtifactStagingDirectory)/onnxruntime-win-x64-gpu-$(OnnxRuntimeVersion).zip'
-
-
-    - template: templates/component-governance-component-detection-steps.yml
-      parameters :
-        condition : 'succeeded'
-
-- stage: NuGet_Packaging_GPU
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:612]: This stage has been moved into `stages/nuget-combine-cuda-stage.yml`

File: tools/ci_build/github/azure-pipelines/cuda-packaging-pipeline.yml [hunk 7]
```diff
@@ -94,29 +94,7 @@ resources:
       ref: 5eda9aded5462201e6310105728d33016e637ea7
 
 stages:
-# Set ReleaseVersionSuffix
```
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/cuda-packaging-pipeline.yml:97]: Moved to `stages/setup-variables-stage.yml`

File: tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml:1]: This has been renamed to 
`tools/ci_build/github/azure-pipelines/stages/jobs/linux-gpu-tensorrt-packaging-job.yml`

File: tools/ci_build/github/azure-pipelines/stages/jobs/linux-gpu-tensorrt-packaging-job.yml
Comment (jchen351) [inline tools/ci_build/github/azure-pipelines/stages/jobs/linux-gpu-tensorrt-packaging-job.yml:1]: Renamed from
`tools/ci_build/github/azure-pipelines/templates/linux-gpu-tensorrt-packaging-pipeline.yml`

**Review by snnn — APPROVED** [review]
Thanks!

**Pull request merged**